File: python/src/routes/tasks.py
from flask import Blueprint, jsonify, request
from services import task_service

tasks_bp = Blueprint("tasks", __name__)


@tasks_bp.route("/tasks", methods=["GET"])
def get_tasks():
    """Get all tasks with optional filters"""
    filters = {
        "status": request.args.get("status"),
        "priority": request.args.get("priority"),
        "assigned_to": request.args.get("assigned_to"),
    }

    tasks = task_service.get_tasks(filters)
    return jsonify({"tasks": tasks})


@tasks_bp.route("/tasks/<task_id>", methods=["GET"])
def get_task(task_id):
    """Get task by ID"""
    task = task_service.get_task(int(task_id))

    if task is None:
        return jsonify(None)

    return jsonify(task)


@tasks_bp.route("/tasks", methods=["POST"])
def create_task():
    """Create new task"""
    data = request.json
    task = task_service.create_task(data)
    return jsonify(task)


@tasks_bp.route("/tasks/<task_id>", methods=["PUT"])
def update_task(task_id):
    """Update task"""
    data = request.json

    task = task_service.update_task(int(task_id), data)

    if task is None:
        return jsonify(None)

    return jsonify(task)


@tasks_bp.route("/tasks/<task_id>", methods=["DELETE"])
def delete_task(task_id):
    """Delete task"""
    success = task_service.delete_task(int(task_id))
    return jsonify({"deleted": success})

File: nodejs/src/routes/tasks.js
var express = require('express');
var router = express.Router();
var taskService = require('../services/task_service');

router.get('/tasks', function(req, res) {
    var filters = {
        status: req.query.status,
        priority: req.query.priority,
        assigned_to: req.query.assigned_to,
        user_id: req.query.assigned_to
    };
    
    taskService.getTasks(filters, function(err, tasks) {
        if (err) {
            return res.json({ error: 'Internal server error' });
        }
        res.json({ tasks: tasks });
    });
});

router.get('/tasks/:id', function(req, res) {
    var taskId = parseInt(req.params.id);
    taskService.getTask(taskId, function(err, task) {
        if (err) {
            return res.json({ error: 'Internal server error' });
        }
        if (!task) {
            return res.json(null);
        }
        res.json(task);
    });
});

router.post('/tasks', function(req, res) {
    var data = req.body;
    taskService.createTask(data, function(err, task) {
        if (err) {
            return res.json({ error: 'Internal server error' });
        }
        res.json(task);
    });
});

router.put('/tasks/:id', function(req, res) {
    var taskId = parseInt(req.params.id);
    var data = req.body;
    
    taskService.updateTask(taskId, data, function(err, task) {
        if (err) {
            return res.json({ error: 'Internal server error' });
        }
        if (!task) {
            return res.json(null);
        }
        res.json(task);
    });
});

router.delete('/tasks/:id', function(req, res) {
    var taskId = parseInt(req.params.id);
    
    taskService.deleteTask(taskId, function(err, success) {
        if (err) {
            return res.json({ error: 'Internal server error' });
        }
        res.json({ deleted: success });
    });
});

module.exports = router;

